keyboards/kinesis/kint36/config.h

```diff
@@ -0,0 +1,87 @@
+/* Copyright 2020 QMK
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#undef VENDOR_ID
+#define VENDOR_ID 0x1209
+#undef PRODUCT_ID
+#define PRODUCT_ID 0x345C
+#undef DEVICE_VER
+#define DEVICE_VER 0x0001
+#undef MANUFACTURER
+#define MANUFACTURER "https://github.com/stapelberg"
+#undef PRODUCT
+#define PRODUCT "https://github.com/kinx-project/kint"
+
+/* key matrix size */
+#define MATRIX_ROWS 15
+#define MATRIX_COLS 7
+
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
+#define MATRIX_ROW_PINS  \
+    {                    \
+        LINE_PIN8,       \
```

Comment:
These can be changed to use the "normal" pin names now.

Reply:
Repeating my other earlier comment:

I tried defining the short pin names in `quantum/config_common.h`, but defining e.g. `C1`, `C2`, `C5`, `C6` results in compilation errors in `lib/chibios-contrib/os/hal/ports/KINETIS/MK66F18/hal_lld.c` and others:

```
lib/chibios-contrib/os/hal/ports/KINETIS/MK66F18/hal_lld.c: In function 'MK66F18_clock_init':
./lib/chibios-contrib/os/hal/ports/KINETIS/LLD/GPIOv1/hal_pal_lld.h:154:3: error: expected identifier before '(' token
  154 |   ((ioline_t)((uint32_t)(port) | ((uint32_t)(pad)<<20)))
      |   ^
./lib/chibios-contrib/os/hal/boards/PJRC_TEENSY_3_6/board.h:296:29: note: in expansion of macro 'PAL_LINE'
  296 | #define LINE_PIN23          PAL_LINE(TEENSY_PIN23_IOPORT, TEENSY_PIN23)
      |                             ^~~~~~~~
quantum/config_common.h:186:20: note: in expansion of macro 'LINE_PIN23'
  186 | #        define C2 LINE_PIN23
      |                    ^~~~~~~~~~
lib/chibios-contrib/os/hal/ports/KINETIS/MK66F18/hal_lld.c:194:10: note: in expansion of macro 'C2'
  194 |     MCG->C2 = MCG_C2_LOCRE0 | MCG_C2_EREFS0 | MCG_C2_RANGE0_SET(2);
      |          ^~

```

Are you sure we need the short names and can’t use the `LINE_PIN*` names from https://github.com/qmk/ChibiOS-Contrib/blob/e3a3a24047717af33b098266e11c2e4e5102206a/os/hal/boards/PJRC_TEENSY_3_6/board.h#L273 ?

---

Are those the normal pin names you mean?

If no, can you please spell out specifically what your suggestion is?

Reply:
It works if you don't include config_common.h, as I said earlier.

Reply:
Indeed, it seems to compile now.

I’m still confused, though: what actually are the short names? (Arduino pinout?) How do I know which short name corresponds to which Teensy pin?

From the Teensy perspective, https://www.pjrc.com/teensy/card11a_rev3_web.pdf documents the pinout. You can see that all pins (aside from power pins) are just numbered.

Reply:
For that you'd have to look at the datasheet, they are prefixed with `PT`: https://www.pjrc.com/teensy/K66P144M180SF5V2.pdf#page=72

The silkscreen numbering is the Arduino pinout, so it's pretty much useless for QMK and doesn't really correspond to anything in the datasheet either.

Reply:
Got it now, thank you! Done.